### ISSUE TITLE
class NextRequest extends Request { ^ ReferenceError: Request is not defined

### ISSUE THREAD

**Issue by adarsh-drishya** [vercel/next.js#57214]
### Link to the code that reproduces this issue

https://github.com/adarsh-drishya/next-turbo-bug

### To Reproduce

1. npm i
2. npm run dev

### Current vs. Expected behavior

```
class NextRequest extends Request {
                          ^

ReferenceError: Request is not defined
```

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows Server 2019 Datacenter
Binaries:
  Node: 16.16.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.7-canary.14
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: N/A
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Additional context

Breaking to even start the local host in npm run dev or even npm run dev-turbo

**Comment by balazsorban44** [comment]
This is expected based on your reproduction. The latest canary requires Node.js 18, but you seem to be on 16. See https://github.com/vercel/next.js/blob/f154bb831358985b69db83e6da276db24d795718/packages/next/package.json#L320

**Comment by Ln2be** [comment]
Thanks you guys, I had the same problem

**Comment by HovhannesMkoyan** [comment]
@balazsorban44 I'm using node v20.9.0 but still getting the same issue. Does the major have to be 18?

**Comment by Mukhammadali** [comment]
It is working with both 18.17 and 20.9.0 for me. All I had to do was just run npm or yarn install in the same terminal where I changed the node version and then run `npm run dev` or `yarn dev` using that terminal only. Or you should make node 18.17 or 20  default node version and then try installing and run dev command

**Comment by HovhannesMkoyan** [comment]
I have node v20 as my global default version, and whenever/wherever I check for the version it's always `20.9.0`.  What's interesting is that explicitly running `npm run dev` in the terminal actually starts the server, however, when trying to run this script from the Visual Studio Code npm scripts section, it gives the error above. I tried to restart both my Visual Studio Code and my Mac, but still getting the issue above when using the script.

**Comment by ben519** [comment]
@HovhannesMkoyan I'm experiencing the same thing. `npm run dev` works but my script fails. For reference, my scripts look like this

```
{
  "name": "myapp",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
...
}
```

(In case it wasn't clear, the same script worked fine before I upgraded next to 14.0.0)

**Comment by PuttTim** [comment]
> This is expected based on your reproduction. The latest canary requires Node.js 18, but you seem to be on 16. See
> 
> https://github.com/vercel/next.js/blob/f154bb831358985b69db83e6da276db24d795718/packages/next/package.json#L320
> 
> For context: Next.js stopped polyfilling `fetch` in favor of the platform-provided implementation, which is present in Node.js 18+ releases.
> 
> Also not that Node.js has reached end of life, so it's good to upgrade anyway.

A warning would be greatly appreciated when either 1) the error pop ups or 2) setting up next.js.

I understand that 16 has reached end of life but having a _very_ vague error that doesn't point you in any direction is a pretty rough DX. Next 14 is now on stable and the setup command pulls from there so a little bit more verbose error handling would be appreciated.

**Comment by LucasSlepetys** [comment]
> I have node v20 as my global default version, and whenever/wherever I check for the version it's always `20.9.0`. What's interesting is that explicitly running `npm run dev` in the terminal actually starts the server, however, when trying to run this script from the Visual Studio Code npm scripts section, it gives the error above. I tried to restart both my Visual Studio Code and my Mac, but still getting the issue above when using the script.

I am facing the exactly same issue. Have you been able to solve it?

**Comment by HovhannesMkoyan** [comment]
@LucasSlepetys I just had a look at the update by @ben519, where provided a link to [this response](https://stackoverflow.com/questions/44700432/visual-studio-code-to-use-node-version-specified-by-nvm/69702082#69702082). I have tried it and it seems to solve my problem.

**Comment by LucasSlepetys** [comment]
> @LucasSlepetys I just had a look at the update by @ben519, where he provided a link to [this response](https://stackoverflow.com/questions/44700432/visual-studio-code-to-use-node-version-specified-by-nvm/69702082#69702082). I have tried it and it seems to solve my problem.

I tried most of the solutions from the response, however none of them seems to be fixing the problem. Was the solution you used about having an old node version in the computer?

**Comment by gurudewan** [comment]
run `nvm install --lts`

**Comment by smitbarmase** [comment]
Updating node to any version in same terminal I'm running script it worked for me.

**Comment by drobnjak10** [comment]
Open terminal inside project and run: npm i node@lts
This helped me.

**Comment by imRanDan** [comment]
Hey, I just had this issue too. I updated nodeJS not through terminal or VScode but by manually going to their website and installing it there. I hope this helps.

**Comment by jsoto007** [comment]
Thanks for this help!

If anyone needs it, I used this step by step guy to install from the terminal: [step by step guide](https://www.bairesdev.com/blog/change-node-version/)

**Comment by sawaYch** [comment]
better to use nvm use lts version node:
```
nvm use --lts
```

**Comment by PeterGKafkakis** [comment]
Updated with NVM to the latest version "v20.9.0" and it is the only node version i have, but for some reason visual studio code still runs Node.js v17.1.0 and cannot figure out why.

**Comment by jsoto007** [comment]
@PeterGKafkakis, maybe you are using V17 as the default version for VS code. Try running
 nvm use v20.9.0 
and restart the app.

**Comment by ben519** [comment]
@PeterGKafkakis it sounds like you have an old version of node installed, probably in `/usr/local/bin`. If so, try running the following:

```
rm -rf /usr/local/bin/npm
rm -rf /usr/local/bin/node
```

**Comment by PeterGKafkakis** [comment]
> @PeterGKafkakis, maybe you are using V17 as the default version for VS code. Try running nvm use v20.9.0 and restart the app.

Tried this multiple times as well as a reboot but still no luck, highly appreciate your response tho!

**Comment by PeterGKafkakis** [comment]
> @PeterGKafkakis it sounds like you have an old version of node installed, probably in `/usr/local/bin`. If so, try running the following:
> 
> ```
> rm -rf /usr/local/bin/npm
> rm -rf /usr/local/bin/node
> ```

How would i use this command on a windows powershell?

**Comment by rajBopche** [comment]
In my case faulty version of node was the culprit. Please ensure you are using node 18+
In the same terminal where error is shown, verify your node version `node -v`
If it shows anything less than 18, bump node to 18+ version. You can use nvm for this.

**Comment by abdulsamadayoade** [comment]
upgrading to the latest node version fixed it for me

**Comment by trucyCc** [comment]
I solved my problem by using version node 18.16.0
